### PR TITLE
feat: release workflow

### PR DESCRIPTION
**Pull request by onbjerg**
- Adds docs on how PRs are merged etc. and how to release a new version
- Adds docs for end-users on update priorities (concept from Lighthouse)
- Adds a release workflow
- Fixes cross-compilation of jemalloc
- Adds a Makefile

### The Makefile

I added a Makefile based off of Lighthouse's makefile. Previously we talked about using other task runners (e.g. `cargo make`), but I opted to just go with the makefile because:

- `make` is widely available on almost all platforms, and on most (except Windows) it is installed by default (or very easy to get)
- `cargo make` ended up being too verbose because of the TOML config
- Other task runners would require separate installation processes (which is why we considered `cargo make` first - everyone who works on this has `cargo install` available)

In some follow up PRs I think it might be worth moving other things out of the GitHub workflow and into the makefile since it gives us a somewhat reproducible way to run all of the test suite locally in a single command.

### Things left to do

- ~~I am having trouble getting `cross` to compile because of MDBX. I've tried installing various constellations of `clang`, `libclang-dev`, `build-essential`, `pkg-config` etc to no avail. It just won't compile.~~

### Config/Secrets

cc @gakonst we need the following:

- `GPG_PASSPHRASE` and `GPG_SIGNING_KEY` secrets to sign the tarballs
- Once we have the above, we need to add the fingerprint to the draft release created by the release workflow

### Follow ups

- A docker release workflow
- Heavier testing suite (e.g. local testnets and testing on remote infrastructure)

### CODE REVIEW

**Comment by gakonst** [comment]
Both secrets are now configured. The signing key is b64 encoded, so you'll need to decode it as below:

```
- name: Configure GPG Key
  run: |
    echo -n "$GPG_SIGNING_KEY" | base64 --decode | gpg --import
  env:
    GPG_SIGNING_KEY: ${{ secrets.GPG_SIGNING_KEY }}
```

**Comment by onbjerg** [comment]
Rebased

File: docs/release.md [hunk 1]
```diff
@@ -1,35 +1,47 @@
...
 
-After pushing the tag, CI will start building the release, bundling it, and publishing it to package managers and crates.io. (TBD)
+#### Releasing
+
+- [ ] Tag the new commit on main with `vx.y.z` (`git tag vx.y.z SHA`)
+- [ ] Push the tag (`git push origin vx.y.z`)[^1]
+- [ ] Run the release commit on testing infrastructure for 1-3 days to check for inconsistencies and bugs
+  - This testing infrastructure is going to sync and keep up with a live testnet, and includes monitoring of bandwidth, CPU, disk space etc.
 
 > **Note**
```
Comment (gakonst) [inline docs/release.md:31]: We should also add releases for the crates, for library users. I've been using this script in Ethers-rs: https://github.com/gakonst/ethers-rs/blob/master/bin/release. We could also use release-plz: https://github.com/MarcoIeni/release-plz which is recent and seems very promising.

**Review by gakonst — APPROVED** [review]
amazing

**Comment by gakonst** [comment]
Test failure on geth seems like a flake? Not sure, cc @Rjected 
```

running 1 test
test can_peer_with_geth ... Fatal: Error starting protocol stack: listen tcp 127.0.0.1:34385: bind: address already in use
FAILED

failures:

failures:
    can_peer_with_geth

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 10.23s


--- STDERR:              reth-staged-sync::sync can_peer_with_geth ---
thread 'can_peer_with_geth' panicked at 'Timed out waiting for geth to start. Is geth installed?', /home/runner/work/reth/reth/crates/staged-sync/src/test_utils/clique.rs:58:29
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

https://github.com/paradigmxyz/reth/actions/runs/5001732852/jobs/8960791989?pr=2677

File: docs/release.md (hunk 1 above)
Comment (onbjerg) [reply docs/release.md:31]: Let's do that in a follow up

**Comment by onbjerg** [comment]
@gakonst I asked @mattsse about that yesterday too, it seems like a flake since I didn't touch anything related to it

**Comment by mattsse** [comment]
actually, I just found it lol
https://github.com/paradigmxyz/reth/pull/2729

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+          cd artifacts
+          tar -czf reth-${{ needs.extract-version.outputs.VERSION }}-${{ matrix.arch }}.tar.gz reth
+          echo "$GPG_PASSPHRASE" | gpg --passphrase-fd 0 --pinentry-mode loopback --batch -ab reth-${{ needs.extract-version.outputs.VERSION }}-${{ matrix.arch }}.tar.gz
+          mv *tar.gz* ..
+
+      - name: Configure GPG and create artifacts Windows
+        if: startsWith(matrix.arch, 'x86_64-windows')
+        env:
+          GPG_SIGNING_KEY: ${{ secrets.GPG_SIGNING_KEY }}
+          GPG_PASSPHRASE: ${{ secrets.GPG_PASSPHRASE }}
```
Comment (mattsse) [inline .github/workflows/release.yml:118]: cool, we have signing

File: Cross.toml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+[build]
+pre-build = [
```
Comment (mattsse) [inline Cross.toml:2]: TIL

**Review by mattsse — COMMENTED** [review]
Incredible stuff.

I'm really equipped to review this properly, but I was able to follow along and lgtm

**Pull request merged**